extensions/wikia/PortableInfobox/tests/nodes/NodeComparisonTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+class NodeComparisonTest extends WikiaBaseTest {
+	protected function setUp() {
+		$this->setupFile = dirname( __FILE__ ) . '/../../PortableInfobox.setup.php';
+		parent::setUp();
+	}
+
+	/**
+	 * @covers       NodeComparison::getData
+	 * @covers       Node::getDataForChildren
+	 * @covers       Node::getChildNodes
+	 * @dataProvider dataProvider
+	 *
+	 * @param $markup
+	 * @param $params
+	 * @param $expected
+	 */
+	public function testGetData( $markup, $params, $expected ) {
+		$node = \Wikia\PortableInfobox\Parser\Nodes\NodeFactory::newFromXML( $markup, $params );
+
+		$this->assertEquals( $expected, $node->getData() );
+	}
+
+	public function dataProvider() {
```

Comment:
What about tests for an empty node and a non-empty node with empty data element?